packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -347,11 +353,13 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
     this.addSemanticIndexes = true,
     this.semanticIndexCallback = _kDefaultSemanticIndexCallback,
     this.semanticIndexOffset = 0,
+    double? estimateChildHeight,
```

Comment:
Asking the developer to provide this parameter is not ideal, The customscrollview should figure this out if possible.

Reply:
Hi，The estimated height of sliver calculated by customscrollview depends on the firstchild and the size after performlayout to estimate the height. Every sliver will initFirstChild, which is not very reasonable.
At the same time, iOS UITableView provides a similar estimated height parameter to determine the extendScrollHeight of the list. We also want to have a similar function on the flutter side to prevent redundant build operations for widgets under multiple slivers.
` 
    if (firstChild == null) {
      if (!addInitialChild()) {
        // There are no children.
        geometry = SliverGeometry.zero;
        childManager.didFinishLayout();
        return;
      }
    }`